src/Build/Evaluation/LazyItemEvaluator.LazyItemOperation.cs

```diff
@@ -110,14 +110,14 @@ protected void DecorateItemsWithMetadata(ICollection<I> items, ImmutableList<Par
 
                     // Do not expand properties as they have been already expanded by the lazy evaluator upon item operation construction.
                     // Prior to lazy evaluation ExpanderOptions.ExpandAll was used.
-                    const ExpanderOptions metadataExpansionOptions = ExpanderOptions.ExpandMetadata | ExpanderOptions.ExpandItems;
```

Comment:
The bug is that recursive metadata is not expanded, however this line clearly states that metadata should get expanded at evaluation time. It's just properties that don't get expanded, as they have already been expanded previously during parsing time. Since `%(RelativeDir)` is metadata and not a property, the cause of the bug must be somewhere else. I'd rather keep the optimization and find the true cause.